### DISCUSSION TITLE
Prism Source Generator Attributes

### DISCUSSION THREAD

**Discussion by YkTru** [PrismLibrary/Prism#2940] [WPF]
@dansiegel 

Hi, 

Regarding source generators I've discovered your Prism.Magician library which seems to be a much welcome (needed?) *modern way* of improving a lot of Prism code (similar to the actual CommunityToolkit.MVVM approach).

1- I totally understand and agree that one must pay for the author's work, especially when you're making $ based on it; but as a solo explorer not knowing yet if I want to use it later for a other professional project, or if I will even finish the one I'm actually working on (I don't have a team and dont intend to make any $ out of my actual work/exploration (it's an experimental app for academic research in logic/linguistic recquiring a lot of different regions/modules)), is there a way for a personal project to have a *community license*?

2- Otherwise, do you know of any library having similar tools I could use? (especially Prism.Magician analyzers, and *much needed* VM-V and Services easy registration with [Attributes]; for VMBase and observable prop/relaycommand attributes I can use MVVM Community Toolkit)

3- Or best scenario: do you plan to eventually make them officially part of Prism?

 Thanks

**Comment by dansiegel** [comment]
The short answer is no there is will not be a community license for the Prism.Magician anytime soon. What I will say currently is to tune in for my session on tomorrow's live stream event with Telerik as I believe it will help answer your question.

https://www.telerik.com/paddle-boarding-in-maui
